PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/GetSyncPollerOnPollerFluxCheck.java

```diff
@@ -0,0 +1,185 @@
+        private static String ANTI_PATTERN_MESSAGE = "";
+
+        private static final Logger LOGGER = Logger.getLogger(GetSyncPollerOnPollerFluxCheck.class.getName());
+
+
+        // Load the config file
+        static {
+            try {
+                List<String> ruleConfigList = getRuleConfigs();
+
+                // extract the method to check and the anti-pattern message from the config file
+                METHOD_TO_CHECK = ruleConfigList.get(0);
+                ANTI_PATTERN_MESSAGE = ruleConfigList.get(1);
+
+            } catch (IOException e) {
+                LOGGER.log(Level.SEVERE, "Error loading rule data", e);
+            }
+        }
+
+        /**
+         * Constructor to initialize the visitor with the holder and isOnTheFly flag.
+         * @param holder Holder for the problems found by the inspection
+         * @param isOnTheFly Flag to indicate if the inspection is running on the fly
+         */
+        public GetSyncPollerOnPollerFluxVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        /**
+         * Method to visit the method call expressions and check for the use of getSyncPoller() on a PollerFlux.
+         * @param expression Method call expression to visit
+         */
+        @Override
+        public void visitMethodCallExpression(@NotNull PsiMethodCallExpression expression) {
+            super.visitMethodCallExpression(expression);
+
+            // Check if the element is a method call expression
+            if (!(expression instanceof PsiMethodCallExpression)) {
+                return;
+            }
+
+            PsiMethodCallExpression methodCall = expression;
+
+            // Check if the method call is getSyncPoller
+            if (methodCall.getMethodExpression().getReferenceName().startsWith(METHOD_TO_CHECK)) {
+                boolean isAsyncContext = checkIfAsyncContext(methodCall);
+
+                if (isAsyncContext && isAzureClient(methodCall)) {
+                    holder.registerProblem(expression, ANTI_PATTERN_MESSAGE);
+                }
+            }
+        }
+
+        /**
+         * Helper method to check if the method call is within an async context.
+         * This method will check if the method call is on a PollerFlux type.
+         *
+         * @param methodCall Method call expression to check
+         * @return true if the method call is on a reactive type, false otherwise
+         */
+        private boolean checkIfAsyncContext(@NotNull PsiMethodCallExpression methodCall) {
+            PsiExpression expression = methodCall.getMethodExpression().getQualifierExpression();
+
+            // Check if the method call is on a reactive type
+            if (expression == null) {
+                return false;
+            }
+            PsiType type = expression.getType();
+
+            // Check if the type is a reactive type
+            if (type == null) {
```

Comment:
how is type = null deduced to not reactive type?

Reply:
this is a check for null pointers before I get the cannonical text so I don't get an exception

Reply:
yeah thats good, I was asking more so about the comment on L127, why is type == null a check for `Check if the type is a reactive type`?

Reply:
oh I think its the placement of the comment that's an issue -- the block of code line 128 - 138 is to check for a PollerFlux -- I'll move it downwards or have "line 128-138 checks for a PollerFlux"